toolchain/sem_ir/type.h

```diff
@@ -53,7 +53,7 @@ class TypeStore : public Yaml::Printable<TypeStore> {
   auto GetTypeIdForTypeInstId(SemIR::InstId inst_id) const -> SemIR::TypeId;
```

Comment:
Should we have an overload of `GetTypeIdForTypeInstId` that skips the validation steps?